### ISSUE TITLE
coc status

### ISSUE THREAD

**Issue by kthenrique** [neoclide/coc-python#138]
Is there a way to hide the "Python \<version> \<arch>" line from the statusbar when using coc#status? It would be nice to have this as a json config like [coc-spell-checker](https://github.com/iamcco/coc-spell-checker) has:

>         "cSpell.showStatus": {
>           "type": "boolean",
>           "scope": "resource",
>           "default": true,
>           "description": "Display the spell checker status on the status bar."
>         },

[Neo]vim already can recognize file formats and many people already has it on their statusbar. It's a duplicity for me to have it switched on. 

![statusbar](https://user-images.githubusercontent.com/33571735/73091698-43887e00-3edb-11ea-9835-4525a2ce271e.png)

**Comment by Gee19** [comment]
Not exactly the ideal fix but I dropped coc status from my statusline and created a custom function:

```
function! StatusDiagnostic() abort
  let info = get(b:, 'coc_diagnostic_info', {})
  if empty(info) | return '' | endif
  let msgs = []
  if get(info, 'error', 0)
    call add(msgs, 'E' . info['error'])
  endif
  if get(info, 'warning', 0)
    call add(msgs, 'W' . info['warning'])
  endif
  return join(msgs, ' ')
endfunction

" Lightline + Tabline
let g:lightline = {
      \ 'colorscheme': 'onedark',
      \ 'active': {
      \   'left': [ [ 'mode', 'paste' ],
      \             [ 'readonly', 'filename', 'modified', 'statusdiag'] ]
      \ },
      \ 'component_function': {
      \   'statusdiag': 'StatusDiagnostic'
      \ },
      \ }

![Screen Shot 2020-01-29 at 1 49 28 AM](https://user-images.githubusercontent.com/5077424/73334150-e9b8e880-4239-11ea-870c-5f4038f168a0.png)

**Comment by kthenrique** [comment]
> Not exactly the ideal fix but I dropped coc status from my statusline and created a custom function:
> 
> ```
> function! StatusDiagnostic() abort
>   let info = get(b:, 'coc_diagnostic_info', {})
>   if empty(info) | return '' | endif
>   let msgs = []
>   if get(info, 'error', 0)
>     call add(msgs, 'E' . info['error'])
>   endif
>   if get(info, 'warning', 0)
>     call add(msgs, 'W' . info['warning'])
>   endif
>   return join(msgs, ' ')
> endfunction
> 
> " Lightline + Tabline
> let g:lightline = {
>       \ 'colorscheme': 'onedark',
>       \ 'active': {
>       \   'left': [ [ 'mode', 'paste' ],
>       \             [ 'readonly', 'filename', 'modified', 'statusdiag'] ]
>       \ },
>       \ 'component_function': {
>       \   'statusdiag': 'StatusDiagnostic'
>       \ },
>       \ }
> ```
> 
> ![Screen Shot 2020-01-29 at 1 49 28 AM](https://user-images.githubusercontent.com/5077424/73334150-e9b8e880-4239-11ea-870c-5f4038f168a0.png)

yeah, but I still wanna be able to see transient status info, like "searching for interpreter" or "Updating extensions" etc.

Thanks for the suggestion tho.

**Comment by Ch00k** [comment]
I believe this is where the interpreter name gets added to the statusline: https://github.com/neoclide/coc-python/blob/master/src/interpreter/display/index.ts#L69 I don't know (yet) how to make that configurable, or if it is at all possible.
